sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -282,6 +285,27 @@ def test_silhouette_nonzero_diag(dtype):
         silhouette_samples(dists, labels, metric="precomputed")
 
 
+@pytest.mark.parametrize("to_sparse", (csr_matrix, csc_matrix))
+def test_silhouette_sparse_input(to_sparse):
```

Comment:
Currently, this test is only smoke test, because it does not assert anything about the output.

I think we can parameterize `test_silhouette_sparse_implementation` to check both `csr_matrix` and `csc_matrix` works and remove `test_silhouette_sparse_input`.